packages/labs/ssr/src/test/lib/render-lit_test.ts

```diff
@@ -545,6 +552,133 @@ for (const global of [emptyVmGlobal, shimmedVmGlobal]) {
       );
     }
   });
+
+  test('server-only templates', async () => {
+    const {render, trivialServerOnly} = await setup();
+    const result = await render(trivialServerOnly);
+    assert.is(result, `<div>Server only</div>`);
+  });
+
+  test('server-only template with a binding', async () => {
+    const {render, serverOnlyWithBinding} = await setup();
+    const result = await render(serverOnlyWithBinding);
+    assert.is(result, `<div>Server only</div>`);
+  });
+
+  test('server-only template with an array', async () => {
+    const {render, serverOnlyArray} = await setup();
+    const result = await render(serverOnlyArray);
+    assert.is(result, `<div>onetwothree</div>`);
+  });
+
+  test('server-only template inside server-only template', async () => {
+    const {render, serverOnlyInsideServerOnly} = await setup();
+    const result = await render(serverOnlyInsideServerOnly);
+    assert.is(result, `<div>Server only</div>`);
+  });
+
+  test(`server-only template can't render a normal template`, async () => {
```

Comment:
Server only can render a normal template?

Reply:
Ah, yeah! I changed the implementation but not the name